### PR TITLE
[stable/coredns] Update labels to follow best practices 

### PR DESCRIPTION
**Pull request by mrueg**
#### What this PR does / why we need it:
Make sure to follow best practices on labeling

#### Special notes for your reviewer:
This includes https://github.com/helm/charts/pull/12892 (your preference to merge it seperately)
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/coredns/templates/clusterrole.yaml [hunk 1]
```diff
@@ -4,15 +4,15 @@ kind: ClusterRole
...
-    heritage: {{ .Release.Service | quote }}
-    release: {{ .Release.Name | quote }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
     {{- if .Values.isClusterService }}
     k8s-app: {{ .Chart.Name | quote }}
     kubernetes.io/cluster-service: "true"
     kubernetes.io/name: "CoreDNS"
```
Comment (shashidharatd) [inline stable/coredns/templates/clusterrole.yaml:13]: Probably the `kubernetes.io/name` label becomes redundant. better to remove. I think same applies for `k8s-app`

Comment (mrueg) [reply stable/coredns/templates/clusterrole.yaml:13]: I did not want to remove that for now, as I didn't want to break things for folks relying on coredns as a cluster dns service.

**Comment by mrueg** [comment]
@shashidharatd rebased, please have a look!

File: stable/coredns/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: coredns
-version: 1.5.1
-appVersion: 1.5.1
+version: 1.5.2
+appVersion: 1.5.0
```
Comment (shashidharatd) [inline stable/coredns/Chart.yaml:3]: Any reason, why you move back the appVersion? we should be moving ahead. please correct it.

Comment (mrueg) [reply stable/coredns/Chart.yaml:3]: The AppVersion encodes coredns' application version (according to https://github.com/helm/helm/blob/master/docs/charts.md ) . There hasn't been a 1.5.1 release yet https://github.com/coredns/coredns/releases/ and the container itself is still set to 1.5.0 in https://github.com/helm/charts/blob/23737aa553e96b7be2a334f4872fa0525cf61840/stable/coredns/values.yaml#L9 

as I mentioned in https://github.com/helm/charts/pull/13416 I think it was changed by accident

**Review by shashidharatd — COMMENTED** [review]
/lgtm
@mrueg Thanks !

File: stable/coredns/Chart.yaml (hunk 2 above)
Comment (shashidharatd) [reply stable/coredns/Chart.yaml:3]: Thanks for the fix.

**Pull request merged**